Jenkinsfile

```diff
@@ -0,0 +1,257 @@
+        contextSource: [
+            $class: "ManuallyEnteredCommitContextSource",
+            context: context
+        ],
+        errorHandlers: [[
+            $class: "ChangingBuildStatusErrorHandler",
+            result: "UNSTABLE"
+        ]],
+        commitShaSource: [
+            $class: "ManuallyEnteredShaSource",
+            sha: env.GIT_COMMIT_HASH
+        ],
+        statusResultSource: [
+            $class: 'ConditionalStatusResultSource',
+            results: [
+                [
+                    $class: 'AnyBuildResult',
+                    message: message,
+                    state: state
+                ]
+            ]
+        ]
+     ])
+}
+
+
+def unittest(String pythonVersion) {
+    echo "set ${pythonVersion} github status"
+    String buildName = "linux ${pythonVersion} test"
+    setBuildStatus('PENDING', "${buildName}", 'test start')
+    try {
+        if (pythonVersion == "py3") {
+            // create python 3 venv
+            sh """
+            python3 -m venv .py3venv --without-pip
+            . .py3venv/bin/activate
+            curl https://bootstrap.pypa.io/get-pip.py | python
+            id
+            python setup.py install
+            pip install coverage mock
+            pip install -I mbed-ls==1.2.14
+            coverage run -m unittest discover -s test -vvv
+            coverage html
+            coverage xml
+            deactivate
+            """
+        } else {
+            // create python2 venv, do installation and run tests
+            sh """
+            virtualenv --python=../usr/bin/python .py2venv --no-site-packages
+            . .py2venv/bin/activate
+            id
+            python setup.py install
+            pip install coverage mock
+            coverage run -m unittest discover -s test -vvv
+            coverage html
+            coverage xml
+            deactivate
+            """
+        }
+        setBuildStatus('SUCCESS', "${buildName}", "test done")
+    } catch (err) {
+        setBuildStatus('FAILURE', "${buildName}", "test failed")
+        currentBuild.result = 'FAILURE'
+    }
+}
+
+
+def linux_pylint_check() {
+    // execute pylint check shell (only for python2, and linux)
+    sh './run_pylint.sh | tee pylint.log'
+
+    // publish warnings
+    step([
+        $class: 'WarningsPublisher',
+        consoleParsers: [
+            [
+                parserName: 'GNU Make + GNU C Compiler (gcc)'
+            ]
+        ],
+        parserConfigurations: [
+            [
+                parserName: 'PyLint',
+                pattern: '**/pylint.log'
+            ]
+        ],
+        unstableTotalAll: '0',
+        failedTotalAll: '0'
+    ])
+}
+
+
+def winTest(String pythonVersion) {
+    echo "set ${pythonVersion} github status"
+    String buildName = "windows ${pythonVersion} test"
+    setBuildStatus('PENDING', "${buildName}", 'test start')
+    try {
+        if (pythonVersion == "py3") {
+            // create python 3 venv
```

Comment:
@juhhov @jonikula is this the correct way to create py2 and py3 venv for windows?

Feels like in a jenkins workspace you cannot cd to dir which it cannot see, like c:\. (maybe just for windows, linux is fine)


````
11:52:49 [windows] 
11:52:49 [windows] C:\Jenkins\ws\sher_PR-60-RPJGRYMVYMYIIMXQKOTXD\mbed-flasher>cd \ 
[Pipeline] [windows] bat
11:52:49 [windows] [mbed-flasher] Running batch script
11:52:50 [windows] 
11:52:50 [windows] C:\Jenkins\ws\sher_PR-60-RPJGRYMVYMYIIMXQKOTXD\mbed-flasher>c:\Python36\python -m venv c:\Jenkins\ws\sher_PR-60-RPJGRYMVYMYIIMXQKOTXD\mbed-flasher\.py3venv 
[Pipeline] [windows] bat
11:53:15 [windows] [mbed-flasher] Running batch script
11:53:16 [windows] 
11:53:16 [windows] C:\Jenkins\ws\sher_PR-60-RPJGRYMVYMYIIMXQKOTXD\mbed-flasher>cd c:\Jenkins\ws\sher_PR-60-RPJGRYMVYMYIIMXQKOTXD\mbed-flasher 
[Pipeline] [windows] bat
11:53:16 [windows] [mbed-flasher] Running batch script
11:53:17 [windows] 
11:53:17 [windows] C:\Jenkins\ws\sher_PR-60-RPJGRYMVYMYIIMXQKOTXD\mbed-flasher>.py3venv\Scripts\activate
[Pipeline] [windows] step
```

Reply:
`virtualenv --python=/usr/bin/python2.6 <path/to/new/virtualenv/>`

https://stackoverflow.com/questions/1534210/use-different-python-version-with-virtualenv